packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -468,414 +426,315 @@ class _MinuteControl extends StatelessWidget {
+      color: pmSelexted ? selectedColor : unselectedColor,
+      child: InkWell(
+        onTap: Feedback.wrapForTap(() => _setPm(context), context),
+        child: Semantics(
+          selected: pmSelexted,
+          child: Center(
+            child: Text(
+              materialLocalizations.postMeridiemAbbreviation,
+              style: pmStyle,
+              textScaleFactor: buttonTextScaleFactor,
+            ),
+          ),
+        ),
+      ),
+    );
+
+    Widget result;
     switch (orientation) {
       case Orientation.portrait:
-        _layoutHorizontally(size, constraints);
+        const double width = 52.0;
+        result = _DayPeriodInputPadding(
+          minSize: const Size(width, kMinInteractiveDimension * 2),
+          orientation: orientation,
+          child: Container(
+            width: width,
+            height: _kTimePickerHeaderControlHeight,
+            child: Material(
+              clipBehavior: Clip.antiAlias,
+              color: Colors.transparent,
+              shape: shape,
+              child: Column(
+                children: <Widget>[
+                  Expanded(child: amButton),
+                  Divider(color: borderColor, height: 1, thickness: 1),
+                  Expanded(child: pmButton),
+                ],
+              ),
+            ),
+          ),
+        );
         break;
       case Orientation.landscape:
-        _layoutVertically(size, constraints);
+        result = _DayPeriodInputPadding(
+          minSize: const Size(0.0, kMinInteractiveDimension),
+          orientation: orientation,
+          child: Container(
+            height: 40.0,
+            child: Material(
+              clipBehavior: Clip.antiAlias,
+              color: Colors.transparent,
+              shape: shape,
+              child: Row(
+                children: <Widget>[
+                  Expanded(child: amButton),
+                  VerticalDivider(color: borderColor, width: 1, thickness: 1),
+                  Expanded(child: pmButton),
+                ],
+              ),
+            ),
+          ),
+        );
         break;
     }
+    return result;
   }
+}
 
-  void _layoutHorizontally(Size size, BoxConstraints constraints) {
-    final List<_TimePickerHeaderFragment> fragmentsFlattened = <_TimePickerHeaderFragment>[];
-    final Map<_TimePickerHeaderId, Size> childSizes = <_TimePickerHeaderId, Size>{};
-    int pivotIndex = 0;
-    for (int pieceIndex = 0; pieceIndex < format.pieces.length; pieceIndex += 1) {
-      final _TimePickerHeaderPiece piece = format.pieces[pieceIndex];
-      for (final _TimePickerHeaderFragment fragment in piece.fragments) {
-        childSizes[fragment.layoutId] = layoutChild(fragment.layoutId, constraints);
-        fragmentsFlattened.add(fragment);
-      }
+/// A widget to pad the area around the [_DayPeriodControl]'s inner [Material].
+class _DayPeriodInputPadding extends SingleChildRenderObjectWidget {
```

Comment:
This looks quite a lot like _RenderInputPadding in button.dart. In a future PR, maybe we should create a shared public widget for this.

Reply:
Yeah I based it off of the button.dart implementation and it could definitely be extracted in a future PR